v2/reader.go

```diff
@@ -9,13 +9,30 @@ import (
 
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-car/v2/internal/carv1"
+	"golang.org/x/exp/mmap"
 )
 
 // Reader represents a reader of CAR v2.
 type Reader struct {
 	Header Header
 	r      io.ReaderAt
 	roots  []cid.Cid
+	closer io.Closer
+}
+
+func NewReaderFromCARv2File(path string) (*Reader, error) {
```

Comment:
How about `NewReaderMmap`? Every reader API takes a CARv2 file, so the current name doesn't really help the user differentiate it from NewReader. And I imagine this will always do an mmap; if we ever changed it to use os.Open instead, that would be a big change for downstreams, so it probably wouldn't be a good idea anyway.

Reply:
@mvdan I agree.